### PR TITLE
Sync with element-skeleton

### PR DESCRIPTION
**Pull request by limonte**
This change is [Reviewable](https://reviewable.io/reviews/vaadin/vaadin-upload/137)

### CODE REVIEW

**Comment by Saulis** [comment]
Reviewed 8 of 9 files at r1.
Review status: 8 of 9 files reviewed at latest revision, 1 unresolved discussion.

---

*[README.md, line 10 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-upload/137#-KYmiWAOKOq72U3LK7Tt:-KYmiWAOKOq72U3LK7Tu:bd9m693) ([raw file](https://github.com/vaadin/vaadin-upload/blob/229f80dd3f1825cbfa2731d8f3d7123a299bac37/README.md#L10)):*
> ```Markdown
>

**Comment by limonte** [comment]
Review status: 8 of 9 files reviewed at latest revision, 1 unresolved discussion.

---

*[README.md, line 10 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-upload/137#-KYmiWAOKOq72U3LK7Tt:-KYmrZ0kAHOg9Q_rqIk8:b-i28k9) ([raw file](https://github.com/vaadin/vaadin-upload/blob/229f80dd3f1825cbfa2731d8f3d7123a299bac37/README.md#L10)):*
<details><summary><i>Previously, Saulis (Sauli Tähkäpää) wrote…</i></summary><blockquote>

This commented out `<custom-element-demo>` is used to display the inline demos in beta.webcomponents.org – for some reason they're not currently visible though? @manolo any ideas why not?
</blockquote></details>

Reverted, thanks, I didn't know that.

Also, beta.webcomponents.org is added.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-upload/137)*

**Comment by manolo** [comment]
Review status: 8 of 9 files reviewed at latest revision, 2 unresolved discussions.

---

*[vaadin-upload.html, line 574 at r2](https://reviewable.io:443/reviews/vaadin/vaadin-upload/137#-KYrNZomD-z60jilFj6B:-KYrNZonEoDCCKMXsIvs:b503ak5) ([raw file](https://github.com/vaadin/vaadin-upload/blob/bb29de71892a4699590453aad4492c920d0ccb09/vaadin-upload.html#L574)):*
> ```HTML
>       }
>       for (var key in this.headers) {
>         if ({}.hasOwnProperty.call(this.headers, key)) {
> ```

I would use `this.headers.hasOwnProperty(key) ` unless we have a good reason for using the `{} call` otherwise looks good to me

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-upload/137)*

File: vaadin-upload.html [hunk 1]
```diff
@@ -571,7 +571,9 @@
         }
       }
       for (var key in this.headers) {
-        xhr.setRequestHeader(key, this.headers[key]);
+        if (this.headers.hasOwnProperty(key)) {
```
Comment (limonte) [inline vaadin-upload.html]: @manolo :+1:

**Comment by platosha** [comment]
Review status: 7 of 9 files reviewed at latest revision, 3 unresolved discussions.

---

*[vaadin-upload.html, line 574 at r2](https://reviewable.io:443/reviews/vaadin/vaadin-upload/137#-KYrNZomD-z60jilFj6B:-KYsJrsf3gTpU3tv1i4E:b-lgi0h6) ([raw file](https://github.com/vaadin/vaadin-upload/blob/bb29de71892a4699590453aad4492c920d0ccb09/vaadin-upload.html#L574)):*
<details><summary><i>Previously, manolo (Manuel Carrasco Moñino) wrote…</i></summary><blockquote>

I would use `this.headers.hasOwnProperty(key) ` unless we have a good reason for using the `{} call` otherwise looks good to me
</blockquote></details>

Once again, this is redundant, and could be even harmful here in the public API.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-upload/137)*

**Comment by limonte** [comment]
Review status: 7 of 9 files reviewed at latest revision, 3 unresolved discussions.

---

*[vaadin-upload.html, line 574 at r2](https://reviewable.io:443/reviews/vaadin/vaadin-upload/137#-KYrNZomD-z60jilFj6B:-KYsX18fay1XIFrg5GAI:b-lmnxm7) ([raw file](https://github.com/vaadin/vaadin-upload/blob/bb29de71892a4699590453aad4492c920d0ccb09/vaadin-upload.html#L574)):*
<details><summary><i>Previously, platosha (Anton Platonov) wrote…</i></summary><blockquote>

Once again, this is redundant, and could be even harmful here in the public API.
</blockquote></details>

`guard-for-in` rule was removed

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-upload/137)*

**Comment by platosha** [comment]
:lgtm:

---

Reviewed 1 of 1 files at r2, 1 of 1 files at r4.
Review status: all files reviewed at latest revision, 3 unresolved discussions, some commit checks broke.

---

*[README.md, line 3 at r4](https://reviewable.io:443/reviews/vaadin/vaadin-upload/137#-KYsjG3yJDyKQaYuYZE8:-KYsjG3yJDyKQaYuYZE9:b-g0v3md) ([raw file](https://github.com/vaadin/vaadin-upload/blob/58b1a84448d2243130abd5cc8598e640d80ea697/README.md#L3)):*
> ```Markdown
> Bower version [Build status](https://travis-ci.org/vaadin/vaadin-upload)
> [Published on webcomponents.org](https://beta.webcomponents.org/element/vaadin/vaadin-upload)
> [Build Status](https://travis-ci.org/vaadin/vaadin-upload)
> ```

Duplicate “Build Status” badge here:


---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-upload/137#-:-KYskoxClJKcudRHs05L:bnfp4nl)*

**Comment by limonte** [comment]
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[README.md, line 3 at r4](https://reviewable.io:443/reviews/vaadin/vaadin-upload/137#-KYsjG3yJDyKQaYuYZE8:-KYw4g3tqfoQAU4Ep60k:bj3eodx) ([raw file](https://github.com/vaadin/vaadin-upload/blob/58b1a84448d2243130abd5cc8598e640d80ea697/README.md#L3)):*
<details><summary><i>Previously, platosha (Anton Platonov) wrote…</i></summary><blockquote>

Duplicate “Build Status” badge here:


</blockquote></details>

Fixed, thanks!

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-upload/137)*

**Pull request merged**